File: 128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        if (nums.size()<1)return 0;
        int longest = 1;
        int lastele= INT_MIN;
        int count = 1;
        for(int i =0;i<nums.size();i++){
            if(nums[i]-1==lastele){
                lastele = nums[i];
                count++;
            }
            else if (nums[i]!=lastele){
                count =1;
                lastele = nums[i];
            }
            longest = max(longest ,count);
            
        }return longest;
    }
};

File: 128.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length ==0) return 0;

        HashSet<Integer> numset = new HashSet<>();

        for(int num : nums){
            numset.add(num);
        }
        int lcs = 0;
        for(int num : numset){
            if(numset.contains(num-1)) continue;
            else{
                int cs = 1;
                int currnum = num;
                
                while(numset.contains(currnum+1)){
                    currnum++;
                    cs++;
                }
                lcs = Math.max(lcs,cs);
            }
        }
        return lcs;
    }
}
